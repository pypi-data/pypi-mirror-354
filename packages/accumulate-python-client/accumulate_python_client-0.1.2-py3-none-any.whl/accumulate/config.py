# accumulate/config.py

# Define Accumulate network base URLs (without "/v3")
# ACCUMULATE_RPC_URL = "https://testnet.accumulatenetwork.io"  # Default: Testnet

ACCUMULATE_RPC_URL = "https://mainnet.accumulatenetwork.io"  # Default: Testnet

# Optional: Add mainnet and custom options
ACCUMULATE_MAINNET_URL = "https://mainnet.accumulatenetwork.io"
ACCUMULATE_TESTNET_URL = "https://testnet.accumulatenetwork.io"
ACCUMULATE_CUSTOM_URL = None  # Set this if using a custom Accumulate node

# Function to get the active Accumulate RPC URL
def get_accumulate_rpc_url():
    """Returns the currently selected Accumulate RPC URL."""
    return ACCUMULATE_CUSTOM_URL or ACCUMULATE_RPC_URL
