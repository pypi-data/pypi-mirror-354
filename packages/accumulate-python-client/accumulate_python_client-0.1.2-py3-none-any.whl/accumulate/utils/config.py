# accumulate-python-client\accumulate\utils\config.py

import os

class Config:
    """
    Configuration utility for managing environment-specific settings.
    """
    @staticmethod
    def is_testnet() -> bool:
        """
        Dynamically get the testnet status based on the environment variable.
        """
        return os.getenv("ACCUMULATE_IS_TESTNET", "false").lower() == "true"

    @staticmethod
    def initial_acme_oracle() -> float:
        """
        Dynamically get the initial ACME oracle value based on the network type.
        """
        return 5000 if Config.is_testnet() else 0.50

    @staticmethod
    def get_network_type() -> str:
        """
        Get the current network type.
        :return: 'testnet' or 'mainnet'
        """
        return "testnet" if Config.is_testnet() else "mainnet"

    @staticmethod
    def get_initial_oracle_value() -> float:
        """
        Get the initial ACME oracle value for the configured network.
        :return: The oracle value as a float.
        """
        return Config.initial_acme_oracle()
