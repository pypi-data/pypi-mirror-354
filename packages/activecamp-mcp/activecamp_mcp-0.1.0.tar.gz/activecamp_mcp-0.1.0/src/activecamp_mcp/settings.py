from pydantic_settings import BaseSettings, SettingsConfigDict


class Settings(BaseSettings):
    host: str = "0.0.0.0"
    port: int = 8080
    prometheus_port: int = 9090
    log_level: str = "INFO"
    reload: bool = False
    workers: int = 1

    mcp_dir: str = "/etc/fastmcp/"
    mcp_webhook_url: str = ""
    mcp_webhook_extra_data: dict = {}
    ac_api_url: str
    ac_api_token: str

    model_config = SettingsConfigDict(
        env_file=".env", env_file_encoding="utf-8", case_sensitive=False, extra="ignore"
    )


settings = Settings()
