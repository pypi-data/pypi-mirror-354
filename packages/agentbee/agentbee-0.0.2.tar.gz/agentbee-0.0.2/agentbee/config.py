import configparser
from pathlib import Path

CONFIG_DIR = Path.home() / ".agentbee"
CONFIG_FILE = CONFIG_DIR / "config.ini"
CONFIG_SECTION = "DEFAULT"

def get_config_path() -> Path:
    """Returns the path to the config file."""
    return CONFIG_FILE

def save_config(api_key: str, base_url: str, model: str):
    """Saves the configuration to the global config file."""
    CONFIG_DIR.mkdir(exist_ok=True)
    config = configparser.ConfigParser()
    config[CONFIG_SECTION] = {
        'llm_api_key': api_key,
        'llm_base_url': base_url,
        'llm_model': model
    }
    with open(CONFIG_FILE, 'w') as configfile:
        config.write(configfile)
    print(f"âœ… Configuration saved to {CONFIG_FILE}")

def load_config() -> dict:
    if not CONFIG_FILE.exists():
        return {}
    config = configparser.ConfigParser()
    config.read(CONFIG_FILE)
    return {
        'llm_api_key': config.get(CONFIG_SECTION, 'llm_api_key', fallback=''),
        'llm_base_url': config.get(CONFIG_SECTION, 'llm_base_url', fallback=''),
        'llm_model': config.get(CONFIG_SECTION, 'llm_model', fallback='')
    }