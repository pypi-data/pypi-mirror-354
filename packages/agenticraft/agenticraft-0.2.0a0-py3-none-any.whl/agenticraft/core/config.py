"""Configuration management for AgentiCraft.

This module provides centralized configuration management using
pydantic-settings. Configuration can be loaded from environment
variables, .env files, or set programmatically.

Example:
    Loading configuration::

        from agenticraft.core.config import settings

        # Access configuration
        print(settings.openai_api_key)
        print(settings.default_model)

        # Override configuration
        settings.default_temperature = 0.5
"""

from functools import lru_cache
from pathlib import Path
from typing import Any

from pydantic import Field, field_validator
from pydantic_settings import BaseSettings, SettingsConfigDict


class AgentiCraftSettings(BaseSettings):
    """Global settings for AgentiCraft.

    Settings are loaded from (in order of priority):
    1. Environment variables
    2. .env file in current directory
    3. .env file in parent directories
    4. Default values

    All settings can be overridden using environment variables
    with the AGENTICRAFT_ prefix.
    """

    model_config = SettingsConfigDict(
        env_prefix="AGENTICRAFT_",
        env_file=".env",
        env_file_encoding="utf-8",
        case_sensitive=False,
        extra="allow",
    )

    # General settings
    environment: str = Field(
        default="development",
        description="Environment (development, staging, production)",
    )
    debug: bool = Field(default=False, description="Enable debug mode")
    log_level: str = Field(default="INFO", description="Logging level")

    # API Keys
    openai_api_key: str | None = Field(default=None, description="OpenAI API key")
    anthropic_api_key: str | None = Field(default=None, description="Anthropic API key")
    google_api_key: str | None = Field(default=None, description="Google API key")

    # Model defaults
    default_model: str = Field(default="gpt-4", description="Default LLM model")
    default_temperature: float = Field(
        default=0.7, ge=0.0, le=2.0, description="Default temperature for LLM"
    )
    default_max_tokens: int | None = Field(
        default=None, gt=0, description="Default max tokens for LLM"
    )
    default_timeout: int = Field(
        default=30, gt=0, description="Default timeout in seconds"
    )
    default_max_retries: int = Field(default=3, ge=0, description="Default max retries")

    # Provider settings
    openai_base_url: str = Field(
        default="https://api.openai.com/v1", description="OpenAI API base URL"
    )
    anthropic_base_url: str = Field(
        default="https://api.anthropic.com", description="Anthropic API base URL"
    )
    ollama_base_url: str = Field(
        default="http://localhost:11434", description="Ollama base URL"
    )

    # Memory settings
    memory_backend: str = Field(
        default="sqlite", description="Memory backend (sqlite, json, redis)"
    )
    memory_path: Path = Field(
        default=Path("./memory"), description="Path for memory storage"
    )
    conversation_memory_size: int = Field(
        default=10, gt=0, description="Number of conversation turns to keep"
    )

    # Telemetry settings
    telemetry_enabled: bool = Field(default=True, description="Enable telemetry")
    telemetry_service_name: str = Field(
        default="agenticraft", description="Service name for telemetry"
    )
    telemetry_export_endpoint: str | None = Field(
        default=None, description="OTLP endpoint for telemetry export"
    )
    telemetry_sample_rate: float = Field(
        default=1.0, ge=0.0, le=1.0, description="Telemetry sampling rate"
    )

    # Plugin settings
    plugins_enabled: bool = Field(default=True, description="Enable plugin system")
    plugin_dirs: list[Path] = Field(
        default_factory=lambda: [Path("./plugins")],
        description="Directories to load plugins from",
    )

    # MCP settings
    mcp_enabled: bool = Field(default=True, description="Enable Model Context Protocol")
    mcp_servers: list[str] = Field(
        default_factory=list, description="MCP server URLs to connect to"
    )

    # Tool settings
    tool_execution_timeout: int = Field(
        default=10, gt=0, description="Timeout for tool execution in seconds"
    )
    tool_retry_attempts: int = Field(
        default=2, ge=0, description="Number of retry attempts for failed tools"
    )

    # Workflow settings
    workflow_step_timeout: int = Field(
        default=60, gt=0, description="Default timeout for workflow steps in seconds"
    )
    workflow_max_parallel_steps: int = Field(
        default=5, gt=0, description="Maximum parallel steps in workflows"
    )

    # Security settings
    allow_code_execution: bool = Field(
        default=False, description="Allow execution of arbitrary code in tools"
    )
    allowed_domains: set[str] = Field(
        default_factory=set, description="Allowed domains for web requests"
    )

    @field_validator("log_level")
    def validate_log_level(cls, v: str) -> str:
        """Validate log level."""
        valid_levels = {"DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"}
        v_upper = v.upper()
        if v_upper not in valid_levels:
            raise ValueError(f"Invalid log level: {v}. Must be one of {valid_levels}")
        return v_upper

    @field_validator("environment")
    def validate_environment(cls, v: str) -> str:
        """Validate environment."""
        valid_envs = {"development", "staging", "production", "test"}
        v_lower = v.lower()
        if v_lower not in valid_envs:
            raise ValueError(f"Invalid environment: {v}. Must be one of {valid_envs}")
        return v_lower

    @field_validator("memory_backend")
    def validate_memory_backend(cls, v: str) -> str:
        """Validate memory backend."""
        valid_backends = {"sqlite", "json", "redis"}
        v_lower = v.lower()
        if v_lower not in valid_backends:
            raise ValueError(
                f"Invalid memory backend: {v}. Must be one of {valid_backends}"
            )
        return v_lower

    def get_api_key(self, provider: str) -> str | None:
        """Get API key for a provider.

        Args:
            provider: Provider name (openai, anthropic, google)

        Returns:
            API key if available
        """
        key_map = {
            "openai": self.openai_api_key,
            "anthropic": self.anthropic_api_key,
            "google": self.google_api_key,
        }
        return key_map.get(provider.lower())

    def get_base_url(self, provider: str) -> str | None:
        """Get base URL for a provider.

        Args:
            provider: Provider name

        Returns:
            Base URL if available
        """
        url_map = {
            "openai": self.openai_base_url,
            "anthropic": self.anthropic_base_url,
            "ollama": self.ollama_base_url,
        }
        return url_map.get(provider.lower())

    def to_agent_config(self) -> dict[str, Any]:
        """Convert settings to agent configuration.

        Returns:
            Dictionary suitable for Agent initialization
        """
        return {
            "model": self.default_model,
            "temperature": self.default_temperature,
            "max_tokens": self.default_max_tokens,
            "timeout": self.default_timeout,
            "max_retries": self.default_max_retries,
        }

    def to_telemetry_config(self) -> dict[str, Any]:
        """Convert settings to telemetry configuration.

        Returns:
            Dictionary suitable for Telemetry initialization
        """
        return {
            "service_name": self.telemetry_service_name,
            "export_to": self.telemetry_export_endpoint,
            "enabled": self.telemetry_enabled,
            "sample_rate": self.telemetry_sample_rate,
        }

    @property
    def is_production(self) -> bool:
        """Check if running in production."""
        return self.environment == "production"

    @property
    def is_development(self) -> bool:
        """Check if running in development."""
        return self.environment == "development"

    def validate_required_keys(self, providers: list[str]) -> None:
        """Validate that required API keys are present.

        Args:
            providers: List of providers that need keys

        Raises:
            ValueError: If required keys are missing
        """
        missing = []
        for provider in providers:
            if not self.get_api_key(provider):
                missing.append(provider)

        if missing:
            raise ValueError(
                f"Missing API keys for providers: {', '.join(missing)}. "
                f"Set environment variables or update .env file."
            )


@lru_cache
def get_settings() -> AgentiCraftSettings:
    """Get the global settings instance (cached).

    Returns:
        AgentiCraftSettings instance
    """
    return AgentiCraftSettings()


# Global settings instance
settings = get_settings()


def reload_settings() -> AgentiCraftSettings:
    """Reload settings from environment.

    This clears the cache and creates a new settings instance.
    Useful for testing or when environment variables change.

    Returns:
        New AgentiCraftSettings instance
    """
    get_settings.cache_clear()
    global settings
    settings = get_settings()
    return settings


def update_settings(**kwargs: Any) -> None:
    """Update settings programmatically.

    Args:
        **kwargs: Setting values to update

    Example:
        update_settings(default_model="gpt-3.5-turbo", debug=True)
    """
    for key, value in kwargs.items():
        if hasattr(settings, key):
            setattr(settings, key, value)
        else:
            raise ValueError(f"Unknown setting: {key}")
