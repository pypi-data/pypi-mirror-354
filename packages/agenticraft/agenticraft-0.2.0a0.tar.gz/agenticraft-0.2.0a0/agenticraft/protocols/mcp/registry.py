"""MCP tool registry for managing tool registrations.

This module provides a registry for MCP tools that can be shared
between clients and servers.
"""

import logging
from typing import Any

from ...core.exceptions import ToolError, ToolNotFoundError
from ...core.tool import BaseTool
from .types import MCPTool, MCPToolParameter

logger = logging.getLogger(__name__)


class MCPRegistry:
    """Registry for MCP tools.

    This registry manages MCP tool definitions and provides
    methods for tool discovery and validation.
    """

    def __init__(self):
        """Initialize the MCP registry."""
        self._tools: dict[str, MCPTool] = {}
        self._categories: dict[str, set[str]] = {}
        self._adapters: dict[str, BaseTool] = {}

    def register_mcp_tool(self, tool: MCPTool, category: str | None = None) -> None:
        """Register an MCP tool.

        Args:
            tool: MCP tool to register
            category: Optional category for organization
        """
        if tool.name in self._tools:
            logger.warning(f"Overwriting existing MCP tool: {tool.name}")

        self._tools[tool.name] = tool

        if category:
            if category not in self._categories:
                self._categories[category] = set()
            self._categories[category].add(tool.name)

        logger.info(f"Registered MCP tool: {tool.name}")

    def register_agenticraft_tool(
        self, tool: BaseTool, category: str | None = None
    ) -> None:
        """Register an AgentiCraft tool as MCP tool.

        Args:
            tool: AgentiCraft tool to register
            category: Optional category
        """
        # Convert to MCP tool
        mcp_tool = self._convert_to_mcp_tool(tool)
        self.register_mcp_tool(mcp_tool, category)

        # Store adapter
        self._adapters[tool.name] = tool

    def _convert_to_mcp_tool(self, tool: BaseTool) -> MCPTool:
        """Convert AgentiCraft tool to MCP tool."""
        definition = tool.get_definition()

        parameters = []
        for param in definition.parameters:
            mcp_param = MCPToolParameter(
                name=param.name,
                type=param.type,
                description=param.description,
                required=param.required,
                default=param.default,
                enum=param.enum,
            )
            parameters.append(mcp_param)

        return MCPTool(
            name=definition.name,
            description=definition.description,
            parameters=parameters,
        )

    def get_tool(self, name: str) -> MCPTool:
        """Get an MCP tool by name.

        Args:
            name: Tool name

        Returns:
            MCP tool

        Raises:
            ToolNotFoundError: If tool not found
        """
        if name not in self._tools:
            raise ToolNotFoundError(name)
        return self._tools[name]

    def get_adapter(self, name: str) -> BaseTool | None:
        """Get AgentiCraft adapter for a tool.

        Args:
            name: Tool name

        Returns:
            AgentiCraft tool adapter if available
        """
        return self._adapters.get(name)

    def list_tools(self, category: str | None = None) -> list[str]:
        """List tool names.

        Args:
            category: Optional category filter

        Returns:
            List of tool names
        """
        if category:
            return list(self._categories.get(category, set()))
        return list(self._tools.keys())

    def list_categories(self) -> list[str]:
        """List available categories.

        Returns:
            List of category names
        """
        return list(self._categories.keys())

    def get_tools_by_category(self, category: str) -> list[MCPTool]:
        """Get all tools in a category.

        Args:
            category: Category name

        Returns:
            List of MCP tools
        """
        tool_names = self._categories.get(category, set())
        return [self._tools[name] for name in tool_names if name in self._tools]

    def search_tools(self, query: str) -> list[MCPTool]:
        """Search for tools by name or description.

        Args:
            query: Search query

        Returns:
            List of matching tools
        """
        query_lower = query.lower()
        matches = []

        for tool in self._tools.values():
            if (
                query_lower in tool.name.lower()
                or query_lower in tool.description.lower()
            ):
                matches.append(tool)

        return matches

    def validate_tool_call(self, tool_name: str, arguments: dict[str, Any]) -> None:
        """Validate tool call arguments.

        Args:
            tool_name: Tool name
            arguments: Tool arguments

        Raises:
            ToolNotFoundError: If tool not found
            ToolError: If arguments are invalid
        """
        tool = self.get_tool(tool_name)

        # Check required parameters
        required_params = {param.name for param in tool.parameters if param.required}
        provided_params = set(arguments.keys())

        missing = required_params - provided_params
        if missing:
            raise ToolError(f"Missing required parameters for {tool_name}: {missing}")

        # Check parameter types (basic validation)
        for param in tool.parameters:
            if param.name in arguments:
                value = arguments[param.name]

                # Check enum values
                if param.enum and value not in param.enum:
                    raise ToolError(
                        f"Invalid value for {param.name}: {value}. "
                        f"Must be one of: {param.enum}"
                    )

    def export_tools(self) -> dict[str, Any]:
        """Export all tools as JSON-serializable dict.

        Returns:
            Dictionary of tools
        """
        return {
            "tools": [tool.to_json_schema() for tool in self._tools.values()],
            "categories": {cat: list(tools) for cat, tools in self._categories.items()},
        }

    def import_tools(self, data: dict[str, Any]) -> None:
        """Import tools from exported data.

        Args:
            data: Exported tools data
        """
        # Clear existing
        self._tools.clear()
        self._categories.clear()

        # Import tools
        for tool_data in data.get("tools", []):
            tool = MCPTool(**tool_data)
            self._tools[tool.name] = tool

        # Import categories
        for category, tool_names in data.get("categories", {}).items():
            self._categories[category] = set(tool_names)

    def clear(self) -> None:
        """Clear all registered tools."""
        self._tools.clear()
        self._categories.clear()
        self._adapters.clear()

    def __len__(self) -> int:
        """Get number of registered tools."""
        return len(self._tools)

    def __contains__(self, tool_name: str) -> bool:
        """Check if tool is registered."""
        return tool_name in self._tools


# Global registry instance
_global_registry = MCPRegistry()


def get_global_registry() -> MCPRegistry:
    """Get the global MCP registry instance.

    Returns:
        Global MCP registry
    """
    return _global_registry
