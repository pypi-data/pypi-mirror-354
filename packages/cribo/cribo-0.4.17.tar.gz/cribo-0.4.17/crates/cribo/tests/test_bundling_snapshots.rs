#![allow(clippy::disallowed_methods)] // insta macros use unwrap internally

use anyhow::Result;
use std::fs;
use std::path::Path;
use std::process::Command;
use tempfile::TempDir;

use cribo::bundler::Bundler;
use cribo::config::Config;
use cribo::util::get_python_executable;

// Ruff linting integration for cross-validation
use ruff_linter::linter::{ParseSource, lint_only};
use ruff_linter::registry::Rule;
use ruff_linter::settings::{LinterSettings, flags};
use ruff_linter::source_kind::SourceKind;
use ruff_python_ast::PySourceType;

/// Structured execution results for better snapshot formatting
#[derive(Debug)]
#[allow(dead_code)] // Fields are used via Debug trait for snapshots
struct ExecutionResults {
    status: ExecutionStatus,
    stdout: String,
    stderr: String,
}

#[derive(Debug)]
#[allow(dead_code)] // Fields are used via Debug trait for snapshots
enum ExecutionStatus {
    Success,
    Failed(i32),
}

/// Ruff linting results for cross-validation
#[derive(Debug)]
#[allow(dead_code)] // Fields are used via Debug trait for snapshots
struct RuffLintResults {
    f401_violations: Vec<String>, // Unused imports
    f404_violations: Vec<String>, // Late future imports
    other_violations: Vec<String>,
    total_violations: usize,
}

/// Run ruff linting on bundled code to cross-validate import handling
fn run_ruff_lint_on_bundle(bundled_code: &str) -> RuffLintResults {
    // Create settings for multiple import-related rules with both F401 and F404 enabled
    let settings = LinterSettings {
        rules: [Rule::UnusedImport, Rule::LateFutureImport]
            .into_iter()
            .collect(),
        ..LinterSettings::default()
    };

    let path = Path::new("<bundled>.py");
    let source_kind = SourceKind::Python(bundled_code.to_string());

    let result = lint_only(
        path,
        None,
        &settings,
        flags::Noqa::Enabled,
        &source_kind,
        PySourceType::Python,
        ParseSource::None,
    );

    let mut f401_violations = Vec::new();
    let mut f404_violations = Vec::new();
    let mut other_violations = Vec::new();

    for message in &result.messages {
        if let Some(rule) = message.to_rule() {
            let location = message.compute_start_location();
            let violation_info = format!(
                "Line {}: {} - {}",
                location.line.get(),
                rule.noqa_code(),
                message.body()
            );

            match rule {
                Rule::UnusedImport => f401_violations.push(violation_info),
                Rule::LateFutureImport => f404_violations.push(violation_info),
                _ => other_violations.push(violation_info),
            }
        }
    }

    let total_violations = f401_violations.len() + f404_violations.len() + other_violations.len();

    RuffLintResults {
        f401_violations,
        f404_violations,
        other_violations,
        total_violations,
    }
}

/// Generic test that processes all fixture directories in tests/fixtures/bundling
/// Each directory should contain a main.py entry point and will be bundled and executed
#[test]
fn test_all_bundling_fixtures() -> Result<()> {
    let fixtures_dir = Path::new(env!("CARGO_MANIFEST_DIR"))
        .join("tests")
        .join("fixtures")
        .join("bundling");

    // Skip if bundling directory doesn't exist
    if !fixtures_dir.exists() {
        return Ok(());
    }

    // Read all subdirectories in the bundling fixtures directory
    let entries = fs::read_dir(&fixtures_dir)?;
    let mut fixture_names = Vec::new();

    for entry in entries {
        let entry = entry?;
        let path = entry.path();

        if path.is_dir() {
            if let Some(name) = path.file_name().and_then(|n| n.to_str()) {
                fixture_names.push(name.to_string());
            }
        }
    }

    // Sort for deterministic test order
    fixture_names.sort();

    // Process each fixture directory
    for fixture_name in fixture_names {
        test_single_bundling_fixture(&fixtures_dir, &fixture_name)?;
    }

    Ok(())
}

/// Test a single bundling fixture directory
fn test_single_bundling_fixture(fixtures_dir: &Path, fixture_name: &str) -> Result<()> {
    let fixture_path = fixtures_dir.join(fixture_name);
    let main_py_path = fixture_path.join("main.py");

    // Skip if main.py doesn't exist
    if !main_py_path.exists() {
        // Skip silently or use conditional verbose logging
        if std::env::var("RUST_TEST_VERBOSE").is_ok() {
            eprintln!("Skipping {}: no main.py found", fixture_name);
        }
        return Ok(());
    }

    // Create temporary directory for output
    let temp_dir = TempDir::new()?;
    let bundle_path = temp_dir.path().join("bundled.py");

    // Configure bundler
    let config = Config::default();
    let mut bundler = Bundler::new(config);

    // Bundle the fixture
    bundler.bundle(&main_py_path, &bundle_path, false)?;

    // Optionally validate Python syntax before execution
    let python_cmd = get_python_executable();
    let syntax_check = Command::new(&python_cmd)
        .args(["-m", "py_compile"])
        .arg(&bundle_path)
        .output();
    if let Ok(output) = syntax_check {
        if !output.status.success() && std::env::var("RUST_TEST_VERBOSE").is_ok() {
            eprintln!(
                "Warning: Bundled code has syntax errors for fixture {}",
                fixture_name
            );
            eprintln!("Stderr: {}", String::from_utf8_lossy(&output.stderr));
        }
    }

    // Read the bundled code and normalize line endings for cross-platform compatibility
    let bundled_code = fs::read_to_string(&bundle_path)?;

    // Run ruff linting for cross-validation of unused imports elimination
    let ruff_results = run_ruff_lint_on_bundle(&bundled_code);

    // Execute the bundled code with Python and capture output
    // Use Python executable with virtual environment support
    let python_cmd = get_python_executable();

    let python_output = Command::new(&python_cmd)
        .arg(&bundle_path)
        .current_dir(temp_dir.path()) // Set working directory for consistent execution
        .output()
        .map_err(|e| {
            anyhow::anyhow!(
                "Failed to execute Python: {} (command: {} {:?})",
                e,
                python_cmd,
                bundle_path
            )
        })?;

    // Note: For timeout support, we'd need a more complex solution or external crate
    // The standard library doesn't provide built-in timeout for process execution

    // Create separate snapshots using insta's named snapshot feature
    insta::with_settings!({
        snapshot_suffix => fixture_name,
        omit_expression => true
    }, {
        // Snapshot the bundled code
        insta::assert_snapshot!("bundled_code", bundled_code);

        // Create structured execution results snapshot
        let execution_status = if python_output.status.success() {
            ExecutionStatus::Success
        } else {
            ExecutionStatus::Failed(python_output.status.code().unwrap_or(-1))
        };

        let execution_results = ExecutionResults {
            status: execution_status,
            stdout: String::from_utf8_lossy(&python_output.stdout)
                .trim()
                .replace("\r\n", "\n")
                .to_string(),
            stderr: String::from_utf8_lossy(&python_output.stderr)
                .trim()
                .replace("\r\n", "\n")
                .to_string(),
        };

        insta::assert_debug_snapshot!("execution_results", execution_results);

        // Snapshot ruff linting results for cross-validation
        insta::assert_debug_snapshot!("ruff_lint_results", ruff_results);
    });

    Ok(())
}
