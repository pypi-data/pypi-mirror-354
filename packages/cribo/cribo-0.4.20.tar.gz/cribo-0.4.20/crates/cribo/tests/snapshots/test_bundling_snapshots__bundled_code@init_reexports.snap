---
source: crates/cribo/tests/test_bundling_snapshots.rs
---
#!/usr/bin/env python3
# Generated by Cribo - Python Source Bundler
# https://github.com/ophidiarium/cribo

# Preserved imports
import os

# ─ Module: mypackage.data_processor ─
import types
mypackage = types.ModuleType('mypackage')
mypackage.data_processor = types.ModuleType('mypackage.data_processor')
exec('"""Data processing module."""\ndef process_data(data):\n    """Process the input data."""\n    processed = data.copy()\n    processed["processed"] = True\n    return processed', mypackage.data_processor.__dict__)

# ─ Module: mypackage.formatter ─
import types
mypackage.formatter = types.ModuleType('mypackage.formatter')
exec('"""Data formatting module."""\ndef format_data(data):\n    """Format the processed data."""\n    return f"Formatted: {data}"', mypackage.formatter.__dict__)

# ─ Module: mypackage.config ─
import types
mypackage.config = types.ModuleType('mypackage.config')
exec('"""Configuration module."""\nimport os\nclass Config:\n    """Configuration class."""\n\n    def __init__(self):\n        self.DEBUG = os.environ.get("DEBUG", "false").lower() == "true"\n        self.LOG_LEVEL = os.environ.get("LOG_LEVEL", "INFO")\nconfig = Config()', globals(), mypackage.config.__dict__)

# ─ Module: mypackage.utils.helper ─
import types
mypackage.utils = types.ModuleType('mypackage.utils')
mypackage.utils.helper = types.ModuleType('mypackage.utils.helper')
exec('"""Helper utilities."""\ndef helper_function(data):\n    """Helper function for processing data."""\n    return f"Helped: {data}"', mypackage.utils.helper.__dict__)

# ─ Module: mypackage.utils.constants ─
import types
mypackage.utils.constants = types.ModuleType('mypackage.utils.constants')
exec('"""Constants module."""\nMAX_ITEMS = 100\nDEFAULT_VALUE = "default"', mypackage.utils.constants.__dict__)

# ─ Module: mypackage.utils ─
exec('global helper_function, MAX_ITEMS, DEFAULT_VALUE\nhelper_function = mypackage.utils.helper.helper_function\nMAX_ITEMS = mypackage.utils.constants.MAX_ITEMS\nDEFAULT_VALUE = mypackage.utils.constants.DEFAULT_VALUE\n"""Utils subpackage with re-exports."""\n__all__ = ["helper_function", "MAX_ITEMS", "DEFAULT_VALUE"]\nlocals()[\'helper_function\'] = helper_function\nlocals()[\'MAX_ITEMS\'] = MAX_ITEMS\nlocals()[\'DEFAULT_VALUE\'] = DEFAULT_VALUE', {**globals(), 'mypackage': mypackage}, mypackage.utils.__dict__)

# ─ Module: mypackage ─
exec('global process_data, format_data, config, helper_function, DEBUG_MODE\nprocess_data = mypackage.data_processor.process_data\nformat_data = mypackage.formatter.format_data\nconfig = mypackage.config.config\nhelper_function = mypackage.utils.helper_function\n"""\\nPackage initialization with re-exports.\\n\\nThis __init__.py demonstrates the pattern where imports are made but not directly used\\nwithin this file - they are re-exports for the package interface.\\nThese imports should NOT be stripped as unused, even though they don\'t appear\\nto be used within this file itself.\\n"""\n__version__ = "1.0.0"\n__all__ = ["process_data", "format_data", "config", "helper_function", "__version__"]\nDEBUG_MODE = config.DEBUG\nlocals()[\'process_data\'] = process_data\nlocals()[\'format_data\'] = format_data\nlocals()[\'config\'] = config\nlocals()[\'helper_function\'] = helper_function\nlocals()[\'DEBUG_MODE\'] = DEBUG_MODE', {**globals(), 'mypackage': mypackage}, mypackage.__dict__)

# ─ Entry Module: main ─
format_data = mypackage.format_data
process_data = mypackage.process_data
config = mypackage.config
helper_function = mypackage.utils.helper_function
'''\nTest script demonstrating __init__.py re-export preservation.\n\nThis fixture tests that imports in __init__.py files are preserved even if they\nappear "unused" within that file, as they are typically re-exports for the package interface.\n'''
def main():
    """Main function demonstrating usage of re-exported functions."""
    data = {"name": "test", "value": 42}
    processed = process_data(data)
    formatted = format_data(processed)
    result = helper_function(formatted)
    if config.DEBUG:
        print(f"Debug: {result}")
    else:
        print(result)
if __name__ == "__main__":
    main()
